File: 프로그래머스/LV0/코딩테스트 입문/유한소수 판별하기/유한소수 판별하기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <math.h>

int solution(int a, int b)
{
    return fmod((float)a/(float)b * 1000, 1) == 0 ? 1 : 2; //  fmod : c에서의 나머지 연산
}

// #include <stdio.h>
// #include <stdbool.h>
// #include <stdlib.h>
// #include <math.h>

// bool is_prime_number(int x)
// {
//     for(int i = 2; i < (int)pow(x, 0.2) + 1; i++)
//     {
//         if(x % i == 0)
//             return false;
//         return true;
//     }
// }

// int gcd(int m, int n)
// {
//     int t;
//     while (n != 0)
//     {
//         t = m % n;
//         m = n;
//         n = t;
//     }
//     return m;
// }

// int solution(int a, int b) 
// {
//     int prime_factors[1000] = {0,};
//     int pf_idx = 0;
//     int denominator;

//     denominator = b / gcd(a, b);

//     if(denominator == 2)
//         return 1;

//     for(int i = 2; i <= denominator; i++)
//     {
//         if(denominator % i == 0 && is_prime_number(i))
//             prime_factors[pf_idx++] = i;
//     }

//     for(int i = 0; i < pf_idx; i++)
//     {
//         if(prime_factors[i] != 2 && prime_factors[i] != 5)
//             return 2;
//     }
//     return 1;
// }

File: 프로그래머스/LV0/코딩테스트 입문/유한소수 판별하기/유한소수 판별하기.py
def solution(a, b):
    return 1 if a/b * 1000 % 1 == 0 else 2

# from math import gcd, sqrt

# def is_prime_number(x):
#     for i in range(2, int(sqrt(x)) + 1):
#         if x % i == 0:
#             return False
#     return True

# def solution(a, b):
#     prime_factors = []

#     denominator = b // gcd(a,b)
#     if denominator == 2:
#         return 1

#     for i in range(2, denominator + 1):
#         if denominator % i == 0 and is_prime_number(i):
#             prime_factors.append(i)

#     for pf in prime_factors:
#         if pf != 2 and pf != 5:
#             return 2

#     return 1